src/utils/buildMetadata.ts

```diff
@@ -166,13 +153,218 @@ export function buildMetadata(sheets: Sheet[], defaultCC: string) {
         optionSets,
         trackedEntityAttributes,
         trackedEntityTypes,
-        //programs,
+        programSections: buildprogramSections(sheets),
+        programs: buildPrograms(sheets),
+        programStages: buildProgramStages(sheets),
+        programStageSections: buildProgramStageSections(sheets),
         programRules: buildProgramRules(sheets),
         programRuleActions: buildProgramRuleActions(sheets),
         programRuleVariables: buildProgramRuleVariables(sheets),
     };
 }
 
+function buildPrograms(sheets: Sheet[]) {
+    const get = (name: string) => getItems(sheets, name);
+
+    const programs = get("programs");
+    const programSectionsData = get("programSections");
+    const programStagesData = get("programStages");
+    const trackedEntityTypes = get("trackedEntityTypes");
+    const categoryCombos = get("categoryCombos");
+
+    return programs.map(program => {
+        const trackedEntityType = {
+            id: getByName(trackedEntityTypes, program.trackedEntityType)?.id
+        };
+
+        const programStages = programStagesData
+            .filter((programStages) => {
+                return programStages?.program === program.name;
+            }).map(({ id }) => ({ id }));
+
+        const categoryCombo = {
+            id: getByName(categoryCombos, program.categoryCombo)?.id ?? process.env.DEFAULT_CATEGORY_COMBO_ID
+        };
+
+        const style = {
+            color: program.styleColor ?? "",
+            icon: program.styleIcon ?? ""
+        };
+        delete program.styleColor;
+        delete program.styleIcon;
+
```

Comment:
Do we need to put the color and the icons in the spreadsheet? The possible valid values are unclear to me in the spreadsheet, is it nos possible to choose any color for example? Maybe it would be easier to leave these out?

If we have them, the way this is done here, the `program` is modified when calling `buildProgram()`. We try to avoid that, since then the result of the general building of metadata can depend on the order of function calls; or if we call multiple times the same function, the result will not be the same. Instead, we can do in this case something like the `let data = { ...program } as MetadataItem;` as in other cases, and then modify `data` and return it, so the original `program` is not modified.

Reply:
Asked Pablo if these fields are useful or not. Will let you know when he gets back to me.

Reply:
Rarely used in development, removing it.